blog/_posts/2020-1-23-crash-course-kubernetes-nodes.md

```diff
@@ -0,0 +1,168 @@
+This is post #2 of the series, in which we talk about nodes, a.k.a "the things that hold the things
+that run your app."
+
+If you missed [Post 1: Overview](https://blog.tilt.dev/2019/10/16/crash-course-kubernetes
+-overview.html), you might want to start there. Stay tuned for post #3, which will cover pods
+(the things that actually run your app) and services (how pods talk to each other and the
+outside world).
+
+## What is a Node, and Why Should I Care?
+In the above diagram, the blue squares are nodes. A node is just a machine (physical or virtual,
+it doesn’t matter) that runs some containers, and a Kubernetes process or two. There are two
+types of nodes: **worker nodes**, where your code runs, and the **master node**, which is in
+charge of managing the cluster.
+
+Like most other parts of Kubernetes, nodes are easy to ignore and abstract away when your
+cluster is in a happy state. When things start to break, however, it can be really useful to
+understand just what Kubernetes is doing. 
+
+If you start getting errors like “node(s) had taints” and “could not schedule pod”, knowing how
+nodes work, and how pods are assigned to nodes, will help you figure out what’s going on. And
+if things start to go wrong (“Unable to connect to the [API] server”), or if your `kubectl
+apply` didn’t work like you expected it to, an accurate mental model of the master node and
+all the machinery that runs your cluster will be a lifesaver.
+
+### Wait, What’s a Pod?
+Simply put, a pod is one unit of your app---say, a running instance of `shopping_cart_server`.
+There’s a bit more to it than that, of course, but we’ll delve into the intricacies of pods in
+the next post in this series. 
+
+For the discussion of nodes, all you need to know is that a pod runs your app in a container,
+and pods themselves are containers that run on nodes. 
+
+## Anatomy of a Node
+### Master Node
+First we’ll talk about the master node. It’s called the “master node” because it’s in charge of everything.
+
+In Kubernetes, you specify your configuration _declaratively_---you tell Kubernetes the desired
+state of your cluster (number of instances of your app, which scaling method to use, how to set
+up the networking) and Kubernetes makes it so. 
+
+The master node is how: it is both the means by which you specify your desired state, and the
+machinery for making sure that state is maintained.
+
+The desired cluster state lives on the master node in a datastore called **etcd**, [a consistent,
+highly-available key value store](https://etcd.io/). 
+
+The master node runs a number of [**controllers**](https://kubernetes.io/docs/concepts/architecture/controller/).
+For a given type of Kubernetes object, a controller: 
+* Polls the cluster for state relating to that type; 
+* Compares it to the desired state stored in etcd; and if needed,
+* Calls out to the **API server** to make any changes to the cluster. 
```

Comment:
```suggestion
* Calls out to the **API server** (explained below) to make any changes to the cluster. 
```

maaaaybe? or maybe introduce it in a diagram above? but I think we should either, like, declare that it exists, or explicitly say we're gonna talk about it later and not to worry about it now